clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -162,14 +163,17 @@ public void close() throws IOException {
                 flush(netWriteBuffer);
             }
         } catch (IOException ie) {
-            log.warn("Failed to send SSL Close message ", ie);
+            log.debug("Failed to send SSL Close message ", ie);
         } finally {
             try {
                 socketChannel.socket().close();
                 socketChannel.close();
             } finally {
                 key.attach(null);
                 key.cancel();
+                netReadBuffer = null;
```

Comment:
Not sure I see any harm in this, but it doesn't appear to have been part of the initial patch. Do we need it?

Reply:
This was part of the patch here: https://github.com/apache/kafka/commit/cb21bca400359e6b7e61555578b7539f27d9d2ff#diff-baf1679e1f8e214cb0bea0e092de0085R185